### PR TITLE
[new release] xapi-stdext, xapi-stdext-zerocheck, xapi-stdext-unix, xapi-stdext-threads, xapi-stdext-std, xapi-stdext-pervasives, xapi-stdext-encodings and xapi-stdext-date (4.20.0)

### PR DESCRIPTION
**Pull request by psafont**
Xapi's standard library extension

- Project page: <a href="https://github.com/xapi-project/stdext">https://github.com/xapi-project/stdext</a>

##### CHANGES:

 - date: consolidate the types into a single t
 - date: add conversion functions that have semantic meaning, the previous functions containing 'float' and 'string' will be deprecated in a future release.

### CODE REVIEW

**Comment by mseri** [comment]
@kit-ty-kate there is a `int32` failure to compile tests in 4.02 that I don't understand. Maybe it is due to alcotest in that case. I don't think that should prevent merging

File: packages/xapi-stdext-encodings/xapi-stdext-encodings.4.20.0/opam [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+synopsis: "Xapi's standard library extension, Encodings"
+maintainer: ["Xapi project maintainers"]
+authors: ["Jonathan Ludlam"]
+license: "LGPL-2.1-only WITH OCaml-LGPL-linking-exception"
+homepage: "https://github.com/xapi-project/stdext"
+bug-reports: "https://github.com/xapi-project/stdext/issues"
+depends: [
+  "ocaml"
+  "dune" {>= "1.11"}
+  "alcotest" {with-test}
```
Comment (kit-ty-kate) [inline packages/xapi-stdext-encodings/xapi-stdext-encodings.4.20.0/opam:11]:
```suggestion
  "alcotest" {with-test & >= "0.6.0"}
```

**Comment by dinosaure** [comment]
The package can not compile on Alpine 3.14 & 3.15. Do you want to support these distributions or do you want to make your package unavailable on them?

**Comment by mseri** [comment]
Only `xapi-stdex-unix` due to
```
[...]
# (cd _build/default/lib/xapi-stdext-unix && /usr/bin/gcc -O2 -fno-strict-aliasing -fwrapv -pthread -fPIC -D_FILE_OFFSET_BITS=64 -O2 -fno-strict-aliasing -fwrapv -pthread -fPIC -g -I /home/opam/.opam/4.14/lib/ocaml -I /home/opam/.opam/4.14/lib/base64 -I /home/opam/.opam/4.14/lib/fd-send-recv -I /home/opam/.opam/4.14/lib/logs -I /home/opam/.opam/4.14/lib/ocaml/threads -I /home/opam/.opam/4.14/lib/ppx_sexp_conv/runtime-lib -I /home/opam/.opam/4.14/lib/result -I /home/opam/.opam/4.14/lib/rpclib -I /home/opam/.opam/4.14/lib/rpclib/core -I /home/opam/.opam/4.14/lib/rpclib/internals -I /home/opam/.opam/4.14/lib/rpclib/json -I /home/opam/.opam/4.14/lib/rpclib/xml -I /home/opam/.opam/4.14/lib/rresult -I /home/opam/.opam/4.14/lib/seq -I /home/opam/.opam/4.14/lib/sexplib0 -I /home/opam/.opam/4.14/lib/xapi-backtrace -I /home/opam/.opam/4.14/lib/xapi-stdext-pervasives -I /home/opam/.opam/4.14/lib/xmlm -I /home/opam/.opam/4.14/lib/yojson -o blkgetsize_stubs.o -c blkgetsize_stubs.c)
# blkgetsize_stubs.c:34:10: fatal error: linux/fs.h: No such file or directory
#    34 | #include <linux/fs.h>
#       |          ^~~~~~~~~~~~
# compilation terminated.
# File "lib/xapi-stdext-unix/dune", line 7, characters 4-17:
# 7 |     unixext_stubs
#         ^^^^^^^^^^^^^
# (cd _build/default/lib/xapi-stdext-unix && /usr/bin/gcc -O2 -fno-strict-aliasing -fwrapv -pthread -fPIC -D_FILE_OFFSET_BITS=64 -O2 -fno-strict-aliasing -fwrapv -pthread -fPIC -g -I /home/opam/.opam/4.14/lib/ocaml -I /home/opam/.opam/4.14/lib/base64 -I /home/opam/.opam/4.14/lib/fd-send-recv -I /home/opam/.opam/4.14/lib/logs -I /home/opam/.opam/4.14/lib/ocaml/threads -I /home/opam/.opam/4.14/lib/ppx_sexp_conv/runtime-lib -I /home/opam/.opam/4.14/lib/result -I /home/opam/.opam/4.14/lib/rpclib -I /home/opam/.opam/4.14/lib/rpclib/core -I /home/opam/.opam/4.14/lib/rpclib/internals -I /home/opam/.opam/4.14/lib/rpclib/json -I /home/opam/.opam/4.14/lib/rpclib/xml -I /home/opam/.opam/4.14/lib/rresult -I /home/opam/.opam/4.14/lib/seq -I /home/opam/.opam/4.14/lib/sexplib0 -I /home/opam/.opam/4.14/lib/xapi-backtrace -I /home/opam/.opam/4.14/lib/xapi-stdext-pervasives -I /home/opam/.opam/4.14/lib/xmlm -I /home/opam/.opam/4.14/lib/yojson -o unixext_stubs.o -c unixext_stubs.c)
# unixext_stubs.c:26:11: fatal error: linux/fs.h: No such file or directory
#    26 | # include <linux/fs.h>
#       |           ^~~~~~~~~~~~
# compilation terminated.
```
right? We could simply add an `available` flag there

File: packages/xapi-stdext-unix/xapi-stdext-unix.4.20.0/opam [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+bug-reports: "https://github.com/xapi-project/stdext/issues"
+depends: [
+  "ocaml" {< "5.0"}
+  "dune" {>= "1.11"}
+  "base-unix"
+  "fd-send-recv" {>= "2.0.0"}
+  "odoc" {with-doc}
+  "xapi-stdext-pervasives" {= version}
+]
+build: [
```
Comment (mseri) [inline packages/xapi-stdext-unix/xapi-stdext-unix.4.20.0/opam:16]:
```suggestion
available: os-family != "alpine"
build: [
```

**Comment by mseri** [comment]
Thanks

**Pull request merged**